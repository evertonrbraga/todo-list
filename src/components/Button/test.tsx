import { render, screen } from 'project-testing-library'
import { Button } from '.'

describe('<Button />', () => {
  it('should snapshot the Button component', () => {
    const { container } = render(<Button />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should the title renders correctly', () => {
    render(<Button />)
    const title = screen.getByText('Criar')
    expect(title).toBeInTheDocument()
  })

  it('should render the icon correctly', () => {
    render(<Button />)
    const icon = screen.getByLabelText('√çcone com sinal de mais')
    expect(icon).toBeInTheDocument()
  })
})
