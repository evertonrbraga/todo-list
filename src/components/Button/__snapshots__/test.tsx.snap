// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should snapshot the Button component 1`] = `
.c0 {
  display: flex;
  align-items: center;
  gap: 5px;
  font-weight: bold;
  padding: 12px 17px;
  border: none;
  border-radius: 8px;
  color: #D9D9D9;
  background-color: #1E6F9F;
  transition: 0.2s ease;
  cursor: pointer;
}

.c0:hover {
  background-color: #4EA8DE;
}

.c0 p {
  font-size: 1.1rem;
}

<button
  class="c0"
>
  <p>
    Criar
  </p>
  <svg
    aria-label="Ícone com sinal de mais"
    fill="currentColor"
    height="22"
    viewBox="0 0 256 256"
    width="22"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="none"
      height="256"
      width="256"
    />
    <circle
      cx="128"
      cy="128"
      fill="none"
      r="96"
      stroke="currentColor"
      stroke-miterlimit="10"
      stroke-width="16"
    />
    <line
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
      x1="88"
      x2="168"
      y1="128"
      y2="128"
    />
    <line
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
      x1="128"
      x2="128"
      y1="88"
      y2="168"
    />
  </svg>
</button>
`;
